src/main/groovy/org/boozallen/plugins/jte/job/TemplateFlowDefinition.groovy

```diff
@@ -54,11 +57,19 @@ abstract class TemplateFlowDefinition extends FlowDefinition {
 
     @Override
     FlowExecution create(FlowExecutionOwner owner, TaskListener listener, List<? extends Action> actions) throws Exception {
-        // if this is a replay run, just skip initialization
-        for (Action a : actions) {
-            if (a instanceof CpsFlowFactoryAction2) {
-                return ((CpsFlowFactoryAction2) a).create(this, owner, actions)
+        WorkflowRun run = owner.run()
+        ReplayCause replay = run.getCause(ReplayCause)
+        RestartDeclarativePipelineCause restartDeclarative = run.getCause(RestartDeclarativePipelineCause)
+        if(replay || restartDeclarative){
+            WorkflowRun previous = replay ? replay.getOriginal() : restartDeclarative.getOriginal()
```

Comment:
does/should the Spec test both cases for ReplayCause and RestartDeclarativePipeline Cause?
deferring to your judgement

Reply:
it does in `DeclarativeSpec`... which is how i figured out i had to add that `RestartDeclarativePipelineCause` in addition to the ReplayCause.